packages/kit/test/test.js

```diff
@@ -197,7 +197,20 @@ function duplicate(test_fn, config, is_build) {
 
 				if (start) {
 					response = await context.pages.js.goto(context.base + start);
-					await context.pages.js.evaluate(() => window.started);
+
+					// If redirecting, sometimes below error comes. Therefore wait until redirect.
+					// "page.evaluate: Execution context was destroyed, most likely because of a navigation."
+					// MOTE: waitForNavigation is extremely slow so doesn't use.
+					const max = 100;
+					for (let i = 0; i <= max; i++) {
+						try {
+							await context.pages.js.evaluate(() => window.started);
+							break;
+						} catch (e) {
+							if (!(e instanceof Error) || i === max) throw e;
+							await new Promise((resolve) => setTimeout(resolve, 10));
```

Comment:
I didn't understand in the NOTE above where you were suggesting the `waitForNavigation` might be called. I think that this would be a good place to do it though as an alternative to calling `setTimeout`. I'd prefer not to call `setTimeout` if possible

Reply:
Ah Definitely! I should update it!


---
[This is just additional explanation]

If error comes during `await context.pages.js.evaluate(() => window.started);`,
it means already navigation was changed.
So after call `await context.pages.js.waitForNavigation();`,
`await context.pages.js.evaluate(() => window.started);` should be succeed.
(But run a for loop in case of multiple redirects.)